<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CallContractPage</class>
 <widget class="QWidget" name="CallContractPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>690</width>
    <height>499</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="callContractPage">
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="labelTitle">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Read Smart Contract (This is not a tx)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frameContract">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QGridLayout" name="gridLayout">
            <property name="leftMargin">
             <number>12</number>
            </property>
            <property name="horizontalSpacing">
             <number>15</number>
            </property>
            <property name="verticalSpacing">
             <number>12</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="labelContractAddress">
              <property name="text">
               <string>Contract Address</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QValidatedTextEdit" name="textEditInterface">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>65</height>
               </size>
              </property>
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOff</enum>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <property name="spacing">
               <number>5</number>
              </property>
              <item>
               <widget class="QValidatedLineEdit" name="lineEditContractAddress"/>
              </item>
              <item>
               <widget class="QToolButton" name="loadInfoButton">
                <property name="toolTip">
                 <string>Choose from contract book page</string>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="iconSize">
                 <size>
                  <width>22</width>
                  <height>22</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="Line" name="vLine">
                <property name="maximumSize">
                 <size>
                  <width>1</width>
                  <height>22</height>
                 </size>
                </property>
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="pasteAddressButton">
                <property name="toolTip">
                 <string>Paste contract address from clipboard</string>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="iconSize">
                 <size>
                  <width>22</width>
                  <height>22</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="Line" name="vLine2">
                <property name="maximumSize">
                 <size>
                  <width>1</width>
                  <height>22</height>
                 </size>
                </property>
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="saveInfoButton">
                <property name="toolTip">
                 <string>Save contract info</string>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="iconSize">
                 <size>
                  <width>22</width>
                  <height>22</height>
                 </size>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0" alignment="Qt::AlignTop">
             <widget class="QLabel" name="labelInterface">
              <property name="text">
               <string>Interface (ABI)</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBoxFunction">
         <property name="title">
          <string>Function</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="leftMargin">
           <number>15</number>
          </property>
          <property name="topMargin">
           <number>15</number>
          </property>
          <item>
           <widget class="QScrollArea" name="scrollAreaFunction">
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>640</width>
               <height>155</height>
              </rect>
             </property>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBoxOptional">
         <property name="title">
          <string>Optional</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <property name="leftMargin">
           <number>15</number>
          </property>
          <property name="topMargin">
           <number>15</number>
          </property>
          <property name="horizontalSpacing">
           <number>10</number>
          </property>
          <property name="verticalSpacing">
           <number>6</number>
          </property>
          <item row="0" column="1">
           <widget class="AddressField" name="lineEditSenderAddress">
            <property name="minimumSize">
             <size>
              <width>370</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="labelSenderAddress">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>160</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Sender Address</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>9</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>9</number>
     </property>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Expanding</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonClearAll">
       <property name="text">
        <string>Clear All</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonCallContract">
       <property name="text">
        <string>Read Contract</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QValidatedLineEdit</class>
   <extends>QLineEdit</extends>
   <header>qvalidatedlineedit.h</header>
  </customwidget>
  <customwidget>
   <class>QValidatedTextEdit</class>
   <extends>QTextEdit</extends>
   <header>qvalidatedtextedit.h</header>
  </customwidget>
  <customwidget>
   <class>AddressField</class>
   <extends>QComboBox</extends>
   <header>addressfield.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
